File: Unique_Number_of_Occurrences/Unique_Number_of_Occurrences.py



arr = [1,2,2,1,1,3]
def uniqueOccurrences(arr):
    d={}
    for i in arr:
        if i in d:
            d[i] += 1
        else:
            d[i] = 1

    s1 = list(set(d.values()))
    if len(d) == len(s1):
        return True
    return False
    


print(uniqueOccurrences(arr))

File: Unique_Number_of_Occurrences/Unique_Number_of_Occurrences.go
package main

import (
	"fmt"
)

func main() {
	arr := []int{1, 2, 2, 1, 1, 3}
	fmt.Println(uniqueOccurrences(arr))
}

func uniqueOccurrences(arr []int) bool {
	m1 := make(map[int]int)

	for _, v := range arr {
		m1[v]++
	}

	m2 := make(map[int]int)
	for k, v := range m1 {
		m2[v] = k
	}

	return (len(m1) == len(m2))
}
